docker-compose.yml

```diff
@@ -842,9 +842,14 @@ services:
       PYTEST_ARGS:  # inherit
     volumes: *conda-volumes
     command: &python-conda-command
+    # pytest-cython needs an upper pin of 0.3.0 due
+    # to an issue with typing annotiations
+    # https://github.com/lgpage/pytest-cython/issues/67
+    # With 0.3.* bug fix release, the pin can be removed
       ["
         /arrow/ci/scripts/cpp_build.sh /arrow /build &&
         /arrow/ci/scripts/python_build.sh /arrow /build &&
+        pip install pytest-cython~=0.2.2 &&
```

Comment:
Can we use https://github.com/apache/arrow/blob/main/ci/conda_env_python.txt instead?

Reply:
Sure. Will use https://github.com/apache/arrow/blob/26631d7504420ff00a827d40273b589c6d38860f/ci/conda_env_sphinx.txt#L32 and add the pin there.